packages/navigation/src/NavigationMenuSelect.tsx

```diff
@@ -0,0 +1,52 @@
+import type { CollectionBase } from '@react-types/shared';
+import type { ChangeEvent } from 'react';
+
+import { Select } from '@launchpad-ui/form';
+import { useListState } from '@react-stately/list';
+import { useCallback } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import { titlecase } from './utils';
+
+const NavigationMenuSelect = <T extends object>(props: CollectionBase<T>) => {
+  const state = useListState(props);
+  const { pathname } = useLocation();
+  const navigate = useNavigate();
+
+  const onChange = useCallback(
+    (e: ChangeEvent<HTMLSelectElement>) => {
+      const { to, onClick } = [...state.collection][e.target.selectedIndex].props;
+
+      navigate(to);
+      // We need to honor the signature of the `onClick` prop for NavigationItem in a collapsed state:
+      // (e: MouseEvent, state: NavigationState) => void
```

Comment:
Calling attention to this: `NavigationItem` is what consumers render inside `Navigation`, and there is an `onClick` prop that we want to respect even when rendering in the collapsed state. Because the closest thing we have from `Select` is `onChange`, we create a custom mouse event and pass it along to keep the prop signature consistent.

This `MouseEvent` will be missing some properties (e.g. `target`), but I think we can document the difference clearly.